### DISCUSSION TITLE
Some problems about the preservation of model parameters

### DISCUSSION THREAD

**Discussion by machuanfeng** [BindsNET/bindsnet#627] [Q&A] [answered]
Where are the weights of the connections between neurons saved? What do I need to save if I want to save the trained model?

**Comment by machuanfeng** [comment]
Has been solved.
Save these parameters and then load them in.
![image](https://user-images.githubusercontent.com/50911954/231333295-fa962534-1831-484f-8764-87152fa479a0.png)
![image](https://user-images.githubusercontent.com/50911954/231333683-7d4359ce-e4d2-4e99-9830-303cd56df414.png)

**Comment by Hananel-Hazan** [comment, marked as answer]
Great that you found where are the weights located.
About saving the network,  you can use also the `save`, `load`, and `clone` functions that are in BindsNET to do this from the object. For example [here](https://github.com/BindsNET/bindsnet/blob/7ac9c3d9ef23340c1baf52d8820b6f8e26f8cbb5/bindsnet/network/network.py#L161)
